Fix build with musl by using correct uint type names.

This patch is taken from Sabotage Linux, the license statement for patches and
build scripts in Sabotage Linux says:

    To the extent possible under law, Christian Neukirchen has waived
    all copyright and related or neighboring rights to this work.

    http://creativecommons.org/publicdomain/zero/1.0/

Therefore this should be good to include in OpenEmbedded.

Signed-off-by: Paul Barker <paul@paulbarker.me.uk>

Upstream-status: Pending

diff -u ethtool-3.14.org/internal.h ethtool-3.14/internal.h
--- ethtool-3.14.org/internal.h
+++ ethtool-3.14/internal.h
@@ -7,6 +7,7 @@
 #include "ethtool-config.h"
 #endif
 #include <stdio.h>
+#include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/types.h>
@@ -17,16 +18,16 @@
 
 /* ethtool.h expects these to be defined by <linux/types.h> */
 #ifndef HAVE_BE_TYPES
-typedef __uint16_t __be16;
-typedef __uint32_t __be32;
+typedef uint16_t __be16;
+typedef uint32_t __be32;
 typedef unsigned long long __be64;
 #endif
 
 typedef unsigned long long u64;
-typedef __uint32_t u32;
-typedef __uint16_t u16;
-typedef __uint8_t u8;
-typedef __int32_t s32;
+typedef uint32_t u32;
+typedef uint16_t u16;
+typedef uint8_t u8;
+typedef int32_t s32;
 
 #include "ethtool-copy.h"
 #include "net_tstamp-copy.h"
